### ISSUE TITLE
How about using special JSON schema validator to validate?

### ISSUE THREAD

**Issue by zaaack** [carlansley/swagger2#13]
I see something like https://github.com/epoberezkin/ajv , they follow the specification, and can handle complex date type like date, which current swagger2 can't.

**Comment by carlansley** [comment]
Looks interesting, might be worth switching (especially for performance and async deref).

**Comment by johnschult** [comment]
Yes, that would be nice. Currently defining a property as as below fails validation. Using `type: 'string'` fails without `format: 'date-time'` as well.

As is, there is no way to accurately represent a primitive  [Data Type](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#data-types) of `string` with format `date-time`.

```javascript
{
  createdAt: {
    type: 'string',
    format: 'date-time',
    example: '2018-04-14T18:41:13.586Z',
    readOnly: true
  }
}
